src/torcontrol.cpp

```diff
@@ -395,7 +397,7 @@ class TorController
     static void reconnect_cb(evutil_socket_t fd, short what, void *arg);
 };
 
-TorController::TorController(struct event_base* _base, const std::string& _target):
+TorController::TorController(struct event_base* _base, const std::string& _target, const std::string& mode):
```

Comment:
Please make this interface take an `enum` mode, to make the available choices clear in the API.
Conversion between the argument and the enum should be done outside this class at the call site
(for example `TorControlThread()`)